jbmc/src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -446,21 +448,21 @@ refined_string_exprt java_string_library_preprocesst::replace_char_array(
   code_blockt &code)
 {
   // array is *array_pointer
-  dereference_exprt array=
+  const dereference_exprt array =
     checked_dereference(array_pointer, array_pointer.type().subtype());
   // array_data is array_pointer-> data
   const exprt array_data = get_data(array, symbol_table);
-  symbolt sym_char_array = get_fresh_aux_symbol(
+  const symbolt sym_char_array = get_fresh_aux_symbol(
     array_data.type(), "char_array", "char_array", loc, ID_java, symbol_table);
```

Comment:
Must take function_id to prefix the symbol name

Reply:
I propose to add a function that makes sure it's always the case. @peterschrammel Can you have a look at the last commit?